### ISSUE TITLE
[Feature Request] Channel Panel Programmatic Interface

### ISSUE THREAD

**Issue by TournyMasterBotCurse** [justintv/Twitch-API#489]
I would like an API endpoint (gated by oauth permission, channel_editor seems to be the closest match right now, but this might warrant a new permission) that can make a PUT (/POST) request to the channel panels.

This PUT (/POST) would allow a user to create a new channel panel, or update an existing channel panel. (Updating an existing channel panel should require the Twitch ID to remove ambiguity. Dealers choice if a new Twitch ID is generated, or if the Twitch ID gets reused with a new updated_at date)

The channel panel submitted via the API should follow the same rules for channel panels submitted through the web page editor.

Justification:

This would allow for dynamic content generation while still controlling the content source to be static after generation. This would allow panels to be updated (in an automated way) for most recent twitch highlight (or a rotation of highlights), Newest YouTube video, Updated statistics display, in game notifications that support integration with twitch api, and more. Curious about that latest kill streak is for CobaltStreak? How about what day CohhCarnage is on for his streamer challenge? What game is Lirik *Actually* playing when he has it listed as Barbie's Paradise Beach House?

Twitch channel panels are a very important way for Twitch streamers to disseminate information to their audience. Being able to automate some of it will allow for significant enhancements for users.

**Comment by scagood** [comment]
+1

**Comment by sen-yang** [comment]
+1

**Comment by night** [comment]
Duplicate of #37

**Comment by Pyrobolser** [comment]
I talked about it on the Twitch Devs a while ago (November 2015).
You can find the discussion [here](https://discuss.dev.twitch.tv/t/update-twitch-panel-through-twitch-api/3871/5) but as cool as it could be, it seems like it's not scheduled for the moment unfortunately.
